packages/utils/src/ts-eslint/Config.ts

```diff
@@ -0,0 +1,244 @@
+/* eslint-disable  @typescript-eslint/consistent-indexed-object-style,  @typescript-eslint/no-namespace */
+
+import type { Parser as ParserType } from './Parser';
+import type * as ParserOptionsTypes from './ParserOptions';
+import type { Processor as ProcessorType } from './Processor';
+import type {
+  AnyRuleModule,
+  RuleCreateFunction,
+  SharedConfigurationSettings,
+} from './Rule';
+
+/** @internal */
+export namespace SharedConfig {
+  export type Severity = 0 | 1 | 2;
+  export type SeverityString = 'error' | 'off' | 'warn';
+  export type RuleLevel = Severity | SeverityString;
+
+  export type RuleLevelAndOptions = [RuleLevel, ...unknown[]];
+
+  export type RuleEntry = RuleLevel | RuleLevelAndOptions;
+  export type RulesRecord = Partial<Record<string, RuleEntry>>;
+
+  export type GlobalVariableOptionBase = 'off' | 'readonly' | 'writable';
+  export type GlobalVariableOption = GlobalVariableOptionBase | boolean;
+
+  export interface GlobalsConfig {
+    [name: string]: GlobalVariableOption;
+  }
+  export interface EnvironmentConfig {
+    [name: string]: boolean;
+  }
+
+  export type ParserOptions = ParserOptionsTypes.ParserOptions;
+}
+
+export namespace ClassicConfig {
```

Comment:
I was being kind of petty when I called this "classic config" - realistically given that v9 will physically kill them, it probably should be called "legacy config".